### ISSUE TITLE
Missing environment setting in step "Debian | Installing gnupg"

### ISSUE THREAD

**Issue by Gwynnbleid1** [ansible-collections/community.zabbix#622]
##### SUMMARY
Step Debian | Installing gnupg is missing environment settings
It should be: 
```- name: "Debian | Install gpg key"
  apt_key:
    id: "{{ sign_keys[zabbix_short_version][zabbix_agent_distribution_release]['sign_key'] }}"
    url: http://repo.zabbix.com/zabbix-official-repo.key
  environment:
    http_proxy: "{{ zabbix_http_proxy | default(None) | default(omit) }}"
    https_proxy: "{{ zabbix_https_proxy | default(None) | default(omit) }}"
  when:
    - zabbix_repo == "zabbix"
  become: true
  tags:
    - zabbix-agent
    - init
```

and not:
```name: "Debian | Install gpg key"
  apt_key:
    id: "{{ sign_keys[zabbix_short_version][zabbix_agent_distribution_release]['sign_key'] }}"
    url: http://repo.zabbix.com/zabbix-official-repo.key
  when:
    - zabbix_repo == "zabbix"
  become: true
  tags:
    - zabbix-agent
    - init
 ```

I believe that the same problem may be present in Ubuntu and other versions

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

zabbix_agent

##### ANSIBLE VERSION

```paste below
ansible [core 2.12.1]
  config file = /ansible/ansible.cfg
  configured module search path = ['/home/mberkieta/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]
  jinja version = 2.11.3
  libyaml = True
```

##### CONFIGURATION

```paste below
COLLECTIONS_PATHS(/ansible/ansible.cfg) = ['/ansible/collections']
DEFAULT_BECOME(/ansible/ansible.cfg) = False
DEFAULT_JINJA2_EXTENSIONS(/ansible/ansible.cfg) = jinja2.ext.do
DEFAULT_REMOTE_USER(/ansible/ansible.cfg) = root
DEFAULT_ROLES_PATH(/ansible/ansible.cfg) = ['/ansible/roles']
RETRY_FILES_ENABLED(/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT / Zabbix Version

Debian 11

##### STEPS TO REPRODUCE

Try to use zabbix_http_proxy and zabbix_https_proxy to install agent

```yaml
- role: community.zabbix.zabbix_agent
      zabbix_http_proxy: http://localhost:3128
      zabbix_https_proxy: http://localhost:3128
```

##### EXPECTED RESULTS

no errors

##### ACTUAL RESULTS

[Errno 101] Network is unreachable in task

```paste below

```

**Comment by Iodun** [comment]
I noticed the following error when trying to install the Zabbix agent on a Ubuntu system.
Is this the result of the bug described here?
We are not using any proxy.

TASK [community.zabbix.zabbix_agent : Debian | Install gpg key] ****************
fatal: [mailout01]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'dict object' has no attribute '5'\n\nThe error appears to be in '/tmp/awx_1395_9s7ko6oj/requirements_collections/ansible_collections/community/zabbix/roles/zabbix_agent/tasks/Debian.yml': line 26, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: \"Debian | Install gpg key\"\n  ^ here\n"}

**Comment by dj-wasabi** [comment]
Hi @Iodun 

No, that error has nothing to do with the bug. You probably overwrote a variable, maybe `zabbix_version`, `zabbix_agent_version` or `zabbix_short_version` to something that is not accepted.

**Comment by Iodun** [comment]
Hey @dj-wasabi 

Thanks a lot, I finally found the problem.
We used this variable:
zabbix_agent_version: 5.0

But for some reason AWX silently removed the minor version at runtime, which is not valid anymore, because the role needs Major.Minor version.
zabbix_agent_version: 5

I fixed it by using single quotes around '5.0', but I still have no clue where this problem suddenly came from.
This exact playbook had been used for months without any changes.
